runtime/onert/core/include/exec/ExecutionContext.h

```diff
@@ -66,6 +66,8 @@ struct ExecutionOptions
   bool dump_minmax = false;
   bool trace = false;
   bool profile = false;
+  // when true, do not set UserTensor on output IOTensors
+  bool skip_set_output_user_tensor = false;
```

Comment:
Can user set this field to skip output setting or not?

Reply:
If the user sets `NNFW_ENABLE_INTERNAL_OUTPUT_ALLOC`, `internal_output_alloc` field in `CompilerOptions` will be set to `true`. As a result, `skip_set_output_user_tensor` will also be set to t`rue` by line 454 in `nnfw_session.cc` in the current implementation.

Reply:
`ExecutionOptions` is designed for passing inference option to executor. For example, `profile` option can be off as default, but set to on for specific inference if user want, and set off again when user don't need to profile any more.
So `skip_set_output_user_tensor` looks little bit different with other field. Setting `NNFW_ENABLE_INTERNAL_OUTPUT_ALLOC` compile option will create executor for this feature, and user cannot change this feature usage on inference step. IMHO, using `skip_set_output_user_tensor` looks workaround. 
If we need field to represent using internal output tensor, it may be in executor level. Or if we decide to use workaround for easy API implementation, it's better to use `nnfw_session` IMO.

Reply:
I replaced `skip_set_output_user_tensor` with an explicit `is_internal_output_tensor` field in `IODescription` and added `outputTensor()` to Executors.
This makes internal output tensor usage clear at the executor level and avoids misusing `ExecutionOptions` for features that can't change per inference.